File: app/src/main/java/me/gilo/wc/adapter/MenuAdapter.java
package me.gilo.wc.adapter;

import android.app.Activity;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import me.gilo.wc.R;
import me.gilo.wc.adapter.viewholder.MenuViewHolder;

import java.util.HashMap;
import java.util.List;

public class MenuAdapter extends RecyclerView.Adapter<MenuViewHolder> {
    private List<String> titles;

    public MenuAdapter( List<String> titles) {
        this.titles = titles;
    }

    @Override
    public MenuViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new MenuViewHolder(parent.getContext(), LayoutInflater.from(parent.getContext()).inflate(R.layout.single_menu_item, parent, false));
    }

    @Override
    public void onBindViewHolder(MenuViewHolder holder, int position) {
        holder.renderView(titles.get(position));
    }


    @Override
    public int getItemCount() {
        return titles.size() == 0 ? 0 : titles.size();
    }
}

File: app/src/main/java/me/gilo/wc/adapter/MenuAdapter.kt
package me.gilo.wc.adapter

import android.app.Activity
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import me.gilo.wc.R
import me.gilo.wc.adapter.viewholder.MenuViewHolder

import java.util.HashMap

class MenuAdapter(private val titles: List<String>) : RecyclerView.Adapter<MenuViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MenuViewHolder {
        return MenuViewHolder(
            parent.context,
            LayoutInflater.from(parent.context).inflate(R.layout.single_menu_item, parent, false)
        )
    }

    override fun onBindViewHolder(holder: MenuViewHolder, position: Int) {
        holder.renderView(titles[position])
    }


    override fun getItemCount(): Int {
        return if (titles.isEmpty()) 0 else titles.size
    }
}
